tutorials/nservicebus-101/lesson-3/tutorial.md

```diff
@@ -0,0 +1,203 @@
+---
+title: "NServiceBus 101 Lesson 3: Multiple endpoints"
+---
+
+Up until this point, we have constrained our activities to a single endpoint, but this is not how real systems behave. The strength of a messaging system is the ability to run code in multiple processes, on multiple servers, which can all collaborate by exchanging messages.
+
+In this lesson, we'll move our message handler to a different endpoint, and discussing the concepts that go along with running more than one endpoint.
+
+
+## Objectives
+
+By the end of this lesson, you will have learned:
+
+ * How to run multiple endpoints
+ * How to send messages between endpoints
+ * How to control the logical routing
+
+
+## How do I send messages?
+
+We've already shown how an endpoint can "send a message to itself" using the `SendLocal()` method, which is available on the `IEndpointInstance` that we use in the endpoint startup code to create a UI, and also on the `IMessageHandlerContext` that we can access while handling a message.
+
+snippet:SendLocal
+
+Sending a message to another endpoint is exactly the same, we just need to drop the word **Local** from the method name.
+
+snippet:Send
+
+The main difference is that with `SendLocal()`, the destination (local) for the message is already known. So when we call `Send()`, how does NServiceBus know where to send the message?
+
+
+## Logical routing
+
+We could specify where we want the message to go directly in code. There is actually an overload of the `Send()` method that allows us to do this:
+
+snippet:SendDestination
+
+However, doing that in most cases isn't a good idea. This requires each developer to remember where each message is supposed to go and type it in every time that message is sent.
+
+Instead, NServiceBus should be made aware of the configuration, so that whenever a message is sent, the framework will already know exactly where it should be delivered.
+
+This is **logical routing**, the mapping of specific message types to logical endpoints that can process those messages. Each command message should have one logical endpoint that owns that message and can process it.
+
+We say *logical routing* because this is at a logical layer only, which isn't necessarily the same as *physical routing*. Within one *logical* endpoint, there may be many *physical* endpoint instances deployed to multiple servers.
+
+{{NOTE:
```

Comment:
Linked to both. I can't really see making them both includes that are shared on https://docs.particular.net/nservicebus/concept-overview since here the point is just to illustrate their differences, and including the details that page has (like mentioning sagas) would be too much here.